modules/gdscript/editor_templates/VisualShaderNodeCustom/basic.gd

```diff
@@ -0,0 +1,41 @@
+# meta-description: Visual shader's node plugin template
+
+@tool
+extends _BASE_
+class_name VisualShaderNode_CLASS_
+
+func _get_name() -> String:
+	return "_CLASS_"
+
+func _get_category() -> String:
+	return ""
+
+func _get_description() -> String:
+	return ""
+
+func _get_return_icon_type() -> int:
+	return PORT_TYPE_SCALAR
+
+func _get_input_port_count() -> int:
+	return 0
+
+func _get_input_port_name(port : int) -> String:
+	return ""
+
+func _get_input_port_type(port : int) -> int:
+	return PORT_TYPE_SCALAR
+
+func _get_output_port_count() -> int:
+	return 1
+
+func _get_output_port_name(port : int) -> String:
+	return "result"
+
+func _get_output_port_type(port : int) -> int:
+	return PORT_TYPE_SCALAR
+
+func _get_global_code(mode : Shader.Mode) -> String:
+	return ""
+
+func _get_code(input_vars : Array[String], output_vars : Array[String], mode : Shader.Mode, type : VisualShader.Type) -> String:
```

Comment:
I think this should follow the GDScript style guide, so there shouldn't be a space before `:` for the type hints (here and in other methods with parameters): https://docs.godotengine.org/en/stable/getting_started/scripting/gdscript/gdscript_styleguide.html#static-typing
```suggestion
func _get_code(input_vars: Array[String], output_vars: Array[String], mode: Shader.Mode, type: VisualShader.Type) -> String:
```